app-frontend/src/app/components/pages/project/layer/aoi/index.js

```diff
@@ -1,11 +1,286 @@
+        }
+        return 0;
+    }
+
+    formatDateDisplay(date) {
+        return date.length ? this.moment.utc(date).format('LLL') + ' (UTC)' : 'MM/DD/YYYY';
+    }
+
+    openShapeModal(layer) {
+        const modal = this.modalService.open({
+            component: 'rfEnterTokenModal',
+            resolve: {
+                title: () => 'Enter a name for the vector data',
+                token: () => `${this.project.name} - ${layer.name} AOI `
+                    + `- ${this.formatDateDisplay(layer.createdAt)}`
+            }
+        });
+
+        modal.result.then((name) => {
+            this.shapesService
+                .createShape(
+                    this.generateFeatureCollection(
+                        this.getMultipolygonGeom(layer.geometry),
+                        name
+                    )
+                )
+                .then(() => {
+                    this.$state.go('project.layer');
+                })
+                .catch(err => {
+                    this.$window.alert('There is an error adding this layer\'s AOI as vector data.'
+                        + ' Please try again later');
+                    this.$log.error(err);
+                });
+        }).catch(() => {});
+    }
+
+    generateFeatureCollection(geometry, name = '') {
+        return {
+            type: 'FeatureCollection',
+            features: [this.generateFeature(geometry, name)]
+        };
+    }
+
+    generateFeature(geometry, name = '') {
+        let result = {
+            type: 'Feature',
+            properties: {},
+            geometry
+        };
+
+        if (name.length) {
+            result.id = this.uuid4.generate();
+            result.properties = { name };
+        }
+
+        return result;
+    }
+
+    onCancelDrawAoi() {
+        this.$state.go('project.layer');
+    }
+
+    onChangeFilterList(filterId) {
+        this.fetchPage(1, filterId);
+        this.displayAoiSource = this.filterList.find(fl => fl.id === filterId).name;
+    }
+
+    createItemInfo(fl) {
+        if (this.dataType === 'AOI') {
+            return {
+                id: fl.id,
+                name: fl.name,
+                subtext: '',
+                date: fl.createdAt,
+                colorGroupHex: fl.colorGroupHex,
+                geometry: fl.geometry
+            };
+        }
+
+        if (this.dataType === 'VECTOR') {
+            return {
+                id: fl.id,
+                name: fl.properties.name,
+                subtext: '',
+                date: fl.properties.createdAt,
+                colorGroupHex: defaultColor,
+                geometry: fl.geometry
+            };
+        }
+
+        return 0;
+    }
+
+    onSelect(id) {
+        this.selected.clear();
+        if (!this.selected.has(id)) {
+            const item = this.itemList.find(itm => itm.id === id);
+            if (item) {
+                this.selected.set(id, item);
+                if (this.dataType === 'AOI') {
```

Comment:
This `if` and the next one seem to do the same thing. This could be trimmed down.